mozdownload/scraper.py

```diff
@@ -555,6 +636,39 @@ def path_regex(self):
             raise errors.NotFoundError("Specified sub folder cannot be found",
                                        folder)
 
+    """ nightly """
+    def get_checksum_url(self):
+        """Get the checksum url for nightly build."""
+        filename = os.path.split(self.url)[1]
+        filename = re.sub('\.%s$' % DEFAULT_FILE_EXTENSIONS[self.platform], '', filename)
+        filename = re.sub('\.installer$', '', filename)
+        url = urljoin(self.path, filename + '.checksums')
+        return url
+
+    def get_checksum_name(self):
+        """Get the name of the platform in the checksum file."""
+        name = ''
+        if self.platform == 'win64' or self.platform == 'win32':
+            name = 'install/sea/'
+        name = name + self.binary
+        return name
+
+    def parse_checksum_file(self):
+        """Search for the corresponding checksum data in the checksum file."""
+        with open(self.cfilename) as f:
+            name = self.get_checksum_name()
+            # self.logger.info('Nightly: Name inside checksum file: %s' % name)
+
+            self.checksum = ''
+            for line in f:
+                content = line.split()
+                checksum = content[0]
+                # self.logger.info('Content[1]: %s' % content[0])
+                if content[1] == 'sha512' and content[3] == name:
+                    # self.logger.info('SHA512 Checksum: %s' % checksum)
+                    self.checksum = checksum
+                    break
+
```

Comment:
The same code again. Given that all checksum files are identically across build types, please define a single parse method in the basic scraper class.

Bonus point we you can move out checksum related code to its own module, so it will make it easier later to handle checksums for different source locations (eg. archive.mozilla.org, or taskcluster).